circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -58,6 +67,25 @@ public AmazonS3 newInstance(AmazonS3URI uri, S3S3CopierOptions s3s3CopierOptions
     }
   }
 
+  public AmazonS3 newTargetInstance(AmazonS3URI uri, S3S3CopierOptions s3s3CopierOptions) {
```

Comment:
Does this need to be public?

Reply:
It's not in the interface so that's why you've had to cast the factory to the concrete type when you use it in the actual s3s3 copier. Do you really have to do that or can `newInstance` do what is necessary?

Reply:
The role is only used for the target client and theres no way to differentiate between the source and target, so there needs to be separate methods

Reply:
We can either add this method to the interface or instead we can create two beans: TargetClientFactory and SourceClientFactory which can be implemented by this class but with different configuration. For instance this takes the ReplicaHiveConf so this would be the TargetClientFactory. Happy to discuss :)

Reply:
Do we need source and target client separate still ?

If a `assume-role` is provided, we could just use a client with that assumed role coz it will have read access on the source. So, I feel like, this class could just have `newInstance` method and based on `s3s3copierOptions`, different type of client can be returned to be used either as source or target.

I will sit with Jay and think more about it tomorrow morning.